### PR TITLE
User logs in to account

### PR DESCRIPTION
**Pull request by koggs**
### What we learned
- refreshes knowledge on FactoryBot

### CODE REVIEW

File: app/views/layouts/_navbar.html.haml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- if user_signed_in?
+  -# = link_to "Hello #{current_user.name}", root_path
+  = link_to "Edit", edit_user_registration_path
```
Comment (tochman) [inline app/views/layouts/_navbar.html.haml]: Is editing the profile part of this feature?

File: app/views/layouts/application.html.haml [hunk 2]
```diff
@@ -9,5 +9,5 @@
   %body
     %p.notice= notice
     %p.alert= alert
-    = link_to 'Sign Up',  new_user_registration_path
+    = render "layouts/navbar"
```
Comment (tochman) [inline app/views/layouts/application.html.haml]: I would render the navbar above the flash messages

File: features/step_definitions/basic_step.rb [hunk 3]
```diff
@@ -9,3 +9,9 @@
 When('I fill in {string} with {string}') do |field, text|
   fill_in field, with: text
 end
+
+Given("We have the following fields:") do |table|
```
Comment (tochman) [inline features/step_definitions/basic_step.rb]: Fields?

File: features/user_logs_in_to_account.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    | name    | email           | password    |
+    | Maran   | maran@test.com  | 12345678    |
+
+Scenario: User can log in with valid credentials
+  Given I visit the site
+  When I click "Login"
+  And I fill in "Email" with "maran@test.com"
+  And I fill in "Password" with "12345678"
+  And I click "Log in"
+  Then I should see "Signed in successfully."
```
Comment (tochman) [inline features/user_logs_in_to_account.feature:17]: I would add a sad path scenario

File: db/schema.rb [hunk 5]
```diff
@@ -10,11 +10,20 @@
...
   enable_extension "plpgsql"
 
+  create_table "recipes", force: :cascade do |t|
+    t.string "title"
+    t.text "decsription"
+    t.text "ingredients"
+    t.text "directions"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
```
Comment (tochman) [inline db/schema.rb]: Why is this part of the PR? Please stay on top of your git flow. Not a blocker but should not be here if you always make sure to have the latest code.

File: spec/factories/user.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :user do
+    password '12345678'
+    email { "maran@test.com".gsub(/\s/,'.').downcase }
```
Comment (tochman) [inline spec/factories/user.rb]: Please explain. Looks way to complex

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -5,3 +5,9 @@
   it { is_expected.to have_db_column :email }
   it { is_expected.to have_db_column :encrypted_password }
 end
+
+describe 'Factory' do
+  it 'has a valid' do
```
Comment (tochman) [inline spec/models/user_spec.rb]: Please formulate this message so it makes sense

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/views/layouts/_navbar.html.haml (hunk 1 above)
Comment (ghost) [reply app/views/layouts/_navbar.html.haml]: It is not part of this feature right now. We'll comment it out until we need it.

File: features/step_definitions/basic_step.rb (hunk 3 above)
Comment (ghost) [reply features/step_definitions/basic_step.rb]: Can you please clarify what you mean?

File: db/schema.rb (hunk 5 above)
Comment (ghost) [reply db/schema.rb]: We'll be more vigilant in the future

File: spec/factories/user.rb (hunk 6 above)
Comment (ghost) [reply spec/factories/user.rb]: We re-used it from a previous challenge. But we've taken a look and changed it to a simpler one.

File: spec/models/user_spec.rb (hunk 7 above)
Comment (ghost) [reply spec/models/user_spec.rb]: We amended the text to make sense.

File: features/user_logs_in_to_account.feature (hunk 4 above)
Comment (ghost) [reply features/user_logs_in_to_account.feature:17]: We will add the sad path before you merge this.

File: app/views/layouts/_navbar.html.haml [hunk 8]
```diff
@@ -1,7 +1,2 @@
-- if user_signed_in?
-  -# = link_to "Hello #{current_user.name}", root_path
-  -# = link_to "Edit", edit_user_registration_path
-  = link_to 'Sign Out',  destroy_user_session_path, method: :delete
-- else
   = link_to 'Sign Up', new_user_registration_path
```
Comment (tochman) [inline app/views/layouts/_navbar.html.haml]: I think the conditional should stay. How about this: (to display the currently logged in users email address)
```haml
- if user_signed_in?
   %p= "Logged in as: #{current_user.email}"
- else
  = link_to 'Sign Up', new_user_registration_path
  = link_to 'Login', new_user_session_path
```
Another thought I have is the words, "Sign Up" doeas not match with "Login" IMO . Perhaps you guys should discuss that?

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -1,9 +1,22 @@
...
 RSpec.describe User, type: :model do
+  let(:user){FactoryBot.create(:user)}
+  describe 'Database table' do
+    it { is_expected.to have_db_column :email }
+    it { is_expected.to have_db_column :encrypted_password }
+  end
+
+  describe 'Validation' do
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_presence_of(:password) }
```
Comment (tochman) [inline spec/models/user_spec.rb]: Please use "poetry mode", meaning that you omit the parentheses wherever you can. Be consistent.

**Comment by koggs** [comment]
@tochman 
We made some changes

**Review by tochman — APPROVED** [review]

**Pull request merged**